src/audio/module_adapter/module/volume/volume.c

```diff
@@ -1182,6 +1182,15 @@ static int volume_prepare(struct processing_module *mod)
 	int ret;
 	int i;
 
+	if (mod->priv.state >= MODULE_IDLE) {
```

Comment:
@lyakh I think this commit can also fix #6184 by removing this early-return logic, while keeping other modules work as same behavior.

Reply:
you mean as is this PR will not fix #6034 and to make it fix it this change to volume.c should be removed? Then probably this should be done? In its present form this PR shouldn't change behaviour of any existing modules, right? `.prepare()` will now be propagated to the modules, but then all in-tree modules will immediately return since after `.reset()` the module will be in `MODULE_IDLE` state, is that correct?

Reply:
Yes, exactly. And I stated an example of the bad influence to other modules if we only modify the generic code to make .prepare() propagated to modules every time. (https://github.com/thesofproject/sof/pull/6184#issuecomment-1229174345).